lib/services/role.go

```diff
@@ -2515,6 +2546,51 @@ func (set RoleSet) CheckAccessToRule(ctx RuleContext, namespace string, resource
 	})
 }
 
+// GetKubeResources returns allowed and denied list of Kubernetes Resources configured in the RoleSet.
+func (set RoleSet) GetKubeResources(cluster types.KubeCluster) (allowed, denied []types.KubernetesResource) {
+	var (
+		allowedResources []types.KubernetesResource
+		deniedResources  []types.KubernetesResource
+	)
+	getRoleLabels := types.Role.GetKubernetesLabels
+	for _, role := range set {
+		matchLabels, _, err := MatchLabels(getRoleLabels(role, types.Allow), cluster.GetAllLabels())
+		if err != nil {
+			continue
+		}
+		if !matchLabels {
+			continue
+		}
+		allowedResources = append(allowedResources, role.GetKubeResources(types.Allow)...)
+	}
+
+	for _, role := range set {
+		matchLabels, _, err := MatchLabels(getRoleLabels(role, types.Deny), cluster.GetAllLabels())
+		if err != nil {
+			continue
+		}
+		if !matchLabels {
+			continue
+		}
+		deniedResources = append(deniedResources, role.GetKubeResources(types.Deny)...)
+	}
+
+	return deduplicateKubeResources(allowedResources), deduplicateKubeResources(deniedResources)
+}
+
+func deduplicateKubeResources(sliceList []types.KubernetesResource) []types.KubernetesResource {
+	allKeys := make(map[string]bool)
+	list := make([]types.KubernetesResource, 0, len(sliceList))
```

Comment:
How common are duplicates? How big are these slices?

Should we avoid copying until we know a duplicate exists, or choose a different algorithm? (Ie, is it worth spending time here?)

Reply:
The size shouldn't be too big - at least I hope so - but it will always depend on the user.

As for the existence of duplicates, I would say that they will be common. Multiple roles associated with the user can define the same resources - mainly `Name:*, Namespace:*`.